webapp/src/Service/SubmissionService.php

```diff
@@ -768,4 +768,33 @@ public function getSubmissionZipResponse(Submission $submission): StreamedRespon
 
         return Utils::streamZipFile($tmpfname, 's' . $submission->getSubmitid() . '.zip');
     }
+
+    public function getSubmissionFileResponse(Submission $submission): StreamedResponse
+    {
+        /** @var SubmissionFile[] $files */
+        $files = $submission->getFiles();
+        
+        if (count($files) !== 1) {
+            throw new ServiceUnavailableHttpException(null, 'Submission does not contain exactly one file.');
+        }
+
+        $file = $files[0];
+        $filename = $file->getFilename();
+        $sourceCode = $file->getSourcecode();
+
+        return new StreamedResponse(function () use ($sourceCode) {
+            echo $sourceCode;
+        }, 200, [
+            'Content-Type'        => 'application/octet-stream',
+            'Content-Disposition' => 'attachment; filename="' . $filename . '"',
+            'Content-Length'      => strlen($sourceCode),
+        ]);
+    }
+
+
+    public function getSubmissionFileNums(Submission $submission): int
```

Comment:
nit: I would prefer the name `getSubmissionFileCount` (and I assume we can mark it as private)

Reply:
Ok, Should I modify it once more, or will you handle the changes when merging?